src/generators/json/utils/sections/base.mjs

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+import { enforceArray } from '../../../../utils/array.mjs';
+import { GeneratorError } from '../../../../utils/generator-error.mjs';
+import { transformNodeToString } from '../../../../utils/unist.mjs';
+import { ENTRY_TO_SECTION_TYPE } from '../../constants.mjs';
+
+/**
+ * @param {import('mdast').Heading} header
+ * @param {number} depth
+ * @returns {typeof ENTRY_TO_SECTION_TYPE[string]}
+ */
+function determineType(header) {
+  const fallback = header.depth === 1 ? 'module' : 'text';
+
+  // doc/api/process.md's parent section shouldn't have a defined type, but
+  // it is defined as `global` for whatever reason
+  if (
+    header?.data.slug === 'process' &&
+    header?.data.type === 'global' &&
+    header?.data.depth === 1
+  ) {
+    return 'module';
+  }
+
+  return ENTRY_TO_SECTION_TYPE[header?.data.type ?? fallback];
+}
+
+/**
+ * Adds a description to the section base.
+ * @param {import('../../generated.d.ts').SectionBase} section
+ * @param {Array<import('mdast').RootContent>} nodes
+ */
+export function addDescriptionAndExamples(section, nodes) {
+  nodes.forEach(node => {
+    if (node.type === 'code') {
+      if (section['@example']) {
+        section['@example'] = [
+          ...enforceArray(section['@example']),
+          node.value,
+        ];
+      } else {
+        section['@example'] = node.value;
+      }
+
+      return;
+    }
+
+    // Not code, let's stringify it and add it to the description.
+    section.description ??= '';
+    section.description += `${transformNodeToString(node)}${node.type === 'paragraph' ? '\n' : ' '}`;
+  });
+
+  if (section.description) {
+    section.description = section.description.trim();
+  }
+}
+
+/**
+ * Adds the stability property to the section.
+ * @param {import('../../generated.d.ts').SectionBase} section
+ * @param {import('../../../../utils/buildHierarchy.mjs').HierarchizedEntry} entry
+ */
+export function addStabilityStatus(section, entry) {
+  const stability = entry.stability.children.map(node => node.data)?.[0];
+
+  if (!stability) {
+    return;
+  }
+
+  let value = stability.index;
+  if (typeof value !== 'number') {
+    value = Number(value);
+
+    if (isNaN(value)) {
+      throw new GeneratorError(`Stability index ${stability.index} NaN`);
+    }
+  }
+
+  section.stability = {
+    value,
+    text: stability.description,
+  };
+}
```

Comment:
Can we just do `section.stability = stability`? Like keep the `{ index, description }` format?